src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Repository/LocationMetaRepository.cs

```diff
@@ -226,35 +239,74 @@ private static List<GeographicLevel> ListLocationLevels(IReadOnlySet<string> all
         {
             GeographicLevel.LocalAuthority => new LocationLocalAuthorityOptionMeta
             {
-                PublicId = string.Empty,
                 Label = label,
                 Code = row[LocalAuthorityCsvColumns.NewCode],
                 OldCode = row[LocalAuthorityCsvColumns.OldCode]
             },
             GeographicLevel.School => new LocationSchoolOptionMeta
             {
-                PublicId = string.Empty,
                 Label = label,
                 Urn = row[SchoolCsvColumns.Urn],
                 LaEstab = row[SchoolCsvColumns.LaEstab]
             },
             GeographicLevel.Provider => new LocationProviderOptionMeta
             {
-                PublicId = string.Empty,
                 Label = label,
                 Ukprn = row[ProviderCsvColumns.Ukprn]
             },
-            GeographicLevel.RscRegion => new LocationRscRegionOptionMeta
-            {
-                PublicId = string.Empty,
-                Label = label
-            },
+            GeographicLevel.RscRegion => new LocationRscRegionOptionMeta { Label = label },
             _ => new LocationCodedOptionMeta
             {
-                PublicId = string.Empty,
                 Label = label,
                 Code = row[cols.Codes.First()]
             }
         };
     }
+
+    private async Task<PublicIdMappings> CreatePublicIdMappings(
+        DataSetVersion dataSetVersion,
+        CancellationToken cancellationToken)
+    {
+        var mappings = await EntityFrameworkQueryableExtensions
+            .SingleOrDefaultAsync(publicDataDbContext
+                    .DataSetVersionMappings,
+                mapping => mapping.TargetDataSetVersionId == dataSetVersion.Id,
+                cancellationToken);
+
+        if (mappings is null)
+        {
+            return new PublicIdMappings();
+        }
+
+        var mappingsByLevel = mappings
+            .LocationMappingPlan
```

Comment:
Similar to earlier comment - could select the `LocationMappingPlan` in the query to avoid fetching the rest of the row.

Reply:
Done as part of the other change - ta